PSReadLine/Render.cs

```diff
@@ -980,19 +983,24 @@ private void RecomputeInitialCoords()
 
         private void MoveCursor(int newCursor)
         {
-            // In case the buffer was resized
-            RecomputeInitialCoords();
-            _previousRender.bufferWidth = _console.BufferWidth;
-            _previousRender.bufferHeight = _console.BufferHeight;
-
-            var point = ConvertOffsetToPoint(newCursor);
-            if (point.Y < 0)
+            // Only update screen cursor if the buffer is fully rendered.
+            if (!_waitingToRender)
             {
-                Ding();
-                return;
+                // In case the buffer was resized
+                RecomputeInitialCoords();
+                _previousRender.bufferWidth = _console.BufferWidth;
+                _previousRender.bufferHeight = _console.BufferHeight;
+
+                var point = ConvertOffsetToPoint(newCursor);
+                if (point.Y < 0)
+                {
+                    Ding();
+                    return;
+                }
+
+                _console.SetCursorPosition(point.X, point.Y);
             }
 
-            _console.SetCursorPosition(point.X, point.Y);
             _current = newCursor;
```

Comment:
This should be moved in the `if (!_waitingToRender)` block too, right?

Reply:
You mean the `_current = newCusor;` part?

No.  The insertion cursor needs to be updated, the physical cursor cannot be updated because the render on screen doesn't match what is going on in the buffer.  Not updating the insertion cursor would cause the paste operation to not match the exception report buffer if and when a exception was to occur.  The paste would also definitely result differently, though this would prevent the exceptions related to #1083 during the paste.

This whole issue however, would be much more difficult to reproduce if Terminal wasn't stuffing the `\n` during the paste, but I would think there is always the possibility of this occuring during laggy remote sessions.  I understand the next preview of Terminal will have this fixed.

Reply:
In case `newCursor` results in `point.Y < 0`, we don't set `newCursor` to `_current`. But if that happens when `_waitingToRender == true`, `_current` will be set to `newCursor` unexpectedly.

Reply:
while `_waitingToRender == false` the state of the render is known, and its possible to compute the coords and determine that the physical cursor will be outside the buffer, and then refuse the move request.

While `_waitingToRender == true`, the screen buffer is unknown, and the assumption is that we are receiving a burst of keyboard input and just trying to get it in to the command line buffer, we need to be able to process the current requests of the keys, hoping that when it is time to render, the cursor will be some place acceptable.   If when the render occurs, the cursor will be outside the screen buffer (most likely `point.Y < 0`), then an exception will still occur, but that is basically the same exception as #1083.

In the future, I am hoping that PSReadLine will be able to edit command lines that are larger than the buffer, in which case the test for `point.Y < 0` will execute a re-render to get the cursor back within the screen buffer.  This is the real fix for #1083.

Reply:
I see your point. when rendering is pending, the `_initialX` and `_initialY` are not updated yet, and thus there is no guarantee that `ConvertOffsetToPoint` would produce the right cursor position. Hence, checking `point.Y < 0` is meaningless.
Can you please add some comments above the line `_current = newCursor;` to explain why we are not converting offset to point and check for if it's off the buffer? Also, please resolve the conflict. Thanks!

Reply:
Merge conflict resolved and comment in place.